File: app/src/main/java/com/commonpepper/photosen/ui/fragments/PhotoListFragment.java
package com.commonpepper.photosen.ui.fragments;

import android.os.Bundle;

import com.commonpepper.photosen.ui.viewmodels.PhotoFragmentViewModelFactory;

import androidx.annotation.Nullable;
import androidx.lifecycle.ViewModelProviders;

public class PhotoListFragment extends AbstractListFragment {
    private static final String TAG_DATE = "date";

    public static PhotoListFragment newInstance(String date) {
        PhotoListFragment newFragment = new PhotoListFragment();
        Bundle args = new Bundle();
        args.putString(TAG_DATE, date);
        newFragment.setArguments(args);
        return newFragment;
    }

    public PhotoListFragment() {
    }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        Bundle args = getArguments();
        String order_by = args.getString(TAG_DATE);
        PhotoFragmentViewModelFactory factory = new PhotoFragmentViewModelFactory(order_by);

        mViewModel = ViewModelProviders.of(this, factory).get(PhotoFragmentViewModelFactory.PhotoListFragmentViewModel.class);
    }
}

File: app/src/main/java/com/commonpepper/photosen/ui/fragments/PhotoListFragment.kt
package com.commonpepper.photosen.ui.fragments

import android.os.Bundle
import androidx.lifecycle.ViewModelProviders
import com.commonpepper.photosen.ui.viewmodels.PhotoFragmentViewModelFactory
import com.commonpepper.photosen.ui.viewmodels.PhotoFragmentViewModelFactory.PhotoListFragmentViewModel

class PhotoListFragment : AbstractListFragment() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val args = arguments
        val orderBy = args!!.getString(TAG_DATE) ?: ""
        val factory = PhotoFragmentViewModelFactory(orderBy)
        mViewModel = ViewModelProviders.of(this, factory).get(PhotoListFragmentViewModel::class.java)
    }

    companion object {
        private const val TAG_DATE = "date"
        fun newInstance(date: String?): PhotoListFragment {
            val newFragment = PhotoListFragment()
            val args = Bundle()
            args.putString(TAG_DATE, date)
            newFragment.arguments = args
            return newFragment
        }
    }
}